crash while doing a cut due to q1 out of bounds

Edwood crashes while doing a cut because `t.q1` is way bigger than `len(t.file.b)`.

Full stack trace is here: [crash-20190323-q1.txt](https://github.com/rjkroege/edwood/files/2999657/crash-20190323-q1.txt)

```
panic: runtime error: slice bounds out of range

goroutine 50 [running, locked to thread]:
panic(0x66b880, 0x8f46d0)
        /home/big/go/src/runtime/panic.go:565 +0x2c5 fp=0xc0004b5c60 sp=0xc0004b5bd0 pc=0x42cbf5
runtime.panicslice()
        /home/big/go/src/runtime/panic.go:54 +0xfb fp=0xc0004b5c90 sp=0xc0004b5c60 pc=0x42ba2b
main.(*Buffer).Read(...)
        /home/fhs/go/src/github.com/rjkroege/edwood/buf.go:30
main.cut(0xc0006bc808, 0xc0006bc808, 0x0, 0x101, 0x0, 0x0)
        /home/fhs/go/src/github.com/rjkroege/edwood/exec.go:313 +0x671 fp=0xc0004b7d38 sp=0xc0004b5c90 pc=0x5fcd21
main.(*Text).Select(0xc0006bc808)
        /home/fhs/go/src/github.com/rjkroege/edwood/text.go:1160 +0x3bb fp=0xc0004b7e48 sp=0xc0004b7d38 pc=0x61c98b
main.MovedMouse(0x481, 0x22e, 0x1, 0xa9bc7053)
        /home/fhs/go/src/github.com/rjkroege/edwood/acme.go:409 +0x47d fp=0xc0004b7ea0 sp=0xc0004b7e48 pc=0x5e866d
main.mousethread(0xc00010c000)
        /home/fhs/go/src/github.com/rjkroege/edwood/acme.go:288 +0x2bb fp=0xc0004b7fd8 sp=0xc0004b7ea0 pc=0x5e805b
runtime.goexit()
        /home/big/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc0004b7fe0 sp=0xc0004b7fd8 pc=0x459591
created by main.main
        /home/fhs/go/src/github.com/rjkroege/edwood/acme.go:160 +0x887
```
Delve debugger:
```
$ coredumpctl debug --debugger=dlvcore 24410
<snip>
  Command Line: edwood -a -l /home/fhs/edwood.dump -f /usr/local/plan9/font/lucsans/unicode.10.font -F /usr/local/plan9/font/lucm/unicode.9.font
    Executable: /home/fhs/go/bin/edwood
<snip>
(dlv) bt
 0  0x000000000045aea1 in runtime.raise
    at /home/big/go/src/runtime/sys_linux_amd64.s:150
 1  0x00000000004418db in runtime.dieFromSignal
    at /home/big/go/src/runtime/signal_unix.go:424
 2  0x0000000000441e58 in runtime.sigfwdgo
    at /home/big/go/src/runtime/signal_unix.go:629
 3  0x0000000000440f80 in runtime.sigtrampgo
    at /home/big/go/src/runtime/signal_unix.go:289
 4  0x000000000045b193 in runtime.sigtramp
    at /home/big/go/src/runtime/sys_linux_amd64.s:357
 5  0x00007f06eb4633c0 in ???
    at ?:-1
 6  0x0000000000441a7a in runtime.crash
    at /home/big/go/src/runtime/signal_unix.go:518
 7  0x000000000042d234 in runtime.fatalpanic
    at /home/big/go/src/runtime/panic.go:717
 8  0x000000000042cbf5 in runtime.gopanic
    at /home/big/go/src/runtime/panic.go:565
 9  0x000000000042ba2b in runtime.panicslice
    at /home/big/go/src/runtime/panic.go:54
10  0x00000000005fcd21 in main.(*Buffer).Read
    at ./buf.go:30
11  0x00000000005fcd21 in main.cut
    at ./exec.go:313
12  0x000000000061c98b in main.(*Text).Select
    at ./text.go:1160
13  0x00000000005e866d in main.MovedMouse
    at ./acme.go:409
14  0x00000000005e805b in main.mousethread
    at ./acme.go:288
15  0x0000000000459591 in runtime.goexit
    at /home/big/go/src/runtime/asm_amd64.s:1337
(dlv) up 11
> runtime.raise() /home/big/go/src/runtime/sys_linux_amd64.s:150 (PC: 0x45aea1)
Warning: debugging optimized function
Frame 11: ./exec.go:313 (PC: 5fcd21)
   308:			for q0 < q1 {
   309:				n = q1 - q0
   310:				if n > RBUFSIZE {
   311:					n = RBUFSIZE
   312:				}
=> 313:				t.file.b.Read(q0, r[:n])
   314:				snarfbuf.Insert(snarfbuf.nc(), r[:n])
   315:				q0 += n
   316:			}
   317:			acmeputsnarf()
   318:		}
(dlv) p t.q0 
0
(dlv) p t.q1
3844
(dlv) p t.file.b
main.Buffer len: 997, cap: 1536, [35,32,103,105,116,104,117,98,46,99,111,109,47,107,116,121,101,47,100,117,105,116,100,114,97,119,10,46,47,105,110,105,116,46,103,111,58,49,55,58,49,52,58,32,117,110,100,101,102,105,110,101,100,58,32,110,101,119,87,105,110,100,111,119,...+933 more]
(dlv)
```

--- a/internal/frame/select.go
+++ b/internal/frame/select.go
@@ -147,6 +147,10 @@ func (f *frameimpl) selectimpl(mc *draw.Mousectl, downevent *draw.Mouse, getmore
 			break
 		}
 	}
+	// TODO(fhs): Why is this necessary?
+	if f.sp1 > f.nchars {
+		f.sp1 = f.nchars
+	}
 	return f.sp0, f.sp1
 }
 

